packages/commcare/src/Utils.js

```diff
@@ -0,0 +1,77 @@
+import { composeNextState } from '@openfn/language-common';
+import {
+  request as commonRequest,
+  makeBasicAuthHeader,
+  logResponse,
+} from '@openfn/language-common/util';
+
+/**
+ * @private
+ *  * @function
+ * @param {string} path - Path to resource
+ * @param {object} data - Object which defines data that will be used to create a given instance of resource
+ * @param {string} contentType- Optional field for the header content-type
+ * @param {string} parseAs- Optional for data response parse i.e json
+ */
+export const post = async ({
```

Comment:
Ok, this isn't quite what I was expecting!

The `post` function will never work, because it doesn't have access to auth credentials.

I note that you've refactored Adaptor.js to not use `post` at all, but instead use request. This is fine but not what I expected.

Let's just remove the `post` function entirely, since it's broken and we're not using it :)

Reply:
I tried using the post but the issue was that it has its own state function inside it and the function inside the adaptor. When I run it, I get undefined